apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/UpdateHistory.scala

```diff
@@ -2174,6 +2174,32 @@ class UpdateHistory(
 }
 
 object UpdateHistory {
+
+  // Separate method so we can use this without a full UpdateHistory instance.
+  def getHighestKnownMigrationId(
+      storage: Storage
+  )(implicit
+      ec: ExecutionContext,
+      closeContext: CloseContext,
+      tc: TraceContext,
+  ): Future[Option[Long]] = {
+    storage match {
+      case storage: DbStorage =>
+        for {
+          queryResult <- storage.query(
+            sql"""
+        select max(migration_id)
```

Comment:
What's the query plan for that? It's not obvious to me that this does not do a linear scan which we should avoid.

I would hope that you can instead query the last_ingested_offset table. For that one a linear scan would be less of an issue.

Reply:
Should be ok, we have one per migration id per local party. And it's a query that we only run on startup. And the plan is to remove the "per party" part to scale to more parties.

That said, for this one I'm also less worried about adding an index. Happy either way.